sdk/core/azure_core/src/http/policies/request_instrumentation.rs

```diff
@@ -0,0 +1,716 @@
+                    key: AZ_NAMESPACE_ATTRIBUTE,
+                    value: namespace.into(),
+                });
+            }
+
+            if !request.url().username().is_empty() || request.url().password().is_some() {
+                // If the URL contains a password, we do not log it for security reasons.
+                let full_url = format!(
+                    "{}://{}{}{}{}{}",
+                    request.url().scheme(),
+                    request
+                        .url()
+                        .host()
+                        .map_or_else(|| "unknown_host".to_string(), |h| h.to_string()),
+                    request
+                        .url()
+                        .port()
+                        .map_or_else(String::new, |p| format!(":{}", p)),
+                    request.url().path(),
+                    request
+                        .url()
+                        .query()
+                        .map_or_else(String::new, |q| format!("?{}", q)),
+                    request
+                        .url()
+                        .fragment()
+                        .map_or_else(String::new, |f| format!("#{}", f)),
+                );
+                span_attributes.push(Attribute {
+                    key: URL_FULL_ATTRIBUTE,
+                    value: full_url.into(),
+                });
+            } else {
+                // If no password is present, we log the full URL.
+                span_attributes.push(Attribute {
+                    key: URL_FULL_ATTRIBUTE,
+                    value: request.url().to_string().into(),
+                });
+            }
+
+            if let Some(host) = request.url().host() {
+                span_attributes.push(Attribute {
+                    key: SERVER_ADDRESS_ATTRIBUTE,
+                    value: host.to_string().into(),
+                });
+            }
+            if let Some(port) = request.url().port_or_known_default() {
+                span_attributes.push(Attribute {
+                    key: SERVER_PORT_ATTRIBUTE,
+                    value: port.into(),
+                });
+            }
+            // Get the method as a string to avoid lifetime issues
+            //            let method_str = request.method_as_str();
+            let method_str = request.method().as_str();
+            let span = if let Some(parent_span) = ctx.value::<Arc<dyn Span>>() {
+                // If a parent span exists, start a new span with the parent.
+                tracer.start_span_with_parent(
+                    method_str,
+                    SpanKind::Client,
+                    span_attributes,
+                    parent_span.clone(),
+                )
+            } else {
+                // If no parent span exists, start a new span without a parent.
+                tracer.start_span_with_current(method_str, SpanKind::Client, span_attributes)
+            };
+
+            if let Some(client_request_id) = request
```

Comment:
it's probably a good idea to check if span is recording here and avoid adding any attributes if not (for perf reasons) - https://github.com/open-telemetry/opentelemetry-rust/blob/d59aded375ac82b0a6a99521a1a7973976762469/opentelemetry/src/trace/span.rs#L108